build.gradle.kts

```diff
@@ -0,0 +1,132 @@
+import org.jetbrains.intellij.tasks.RunIdeTask
+import org.jetbrains.intellij.tasks.RunPluginVerifierTask
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import java.util.EnumSet
+
+plugins {
+  id("idea")
+  id("java")
+  kotlin("kapt") version "1.8.10"
+  id("org.jetbrains.intellij") version "1.13.3"
+  id("org.jetbrains.kotlin.jvm") version "1.8.10"
+  id("org.jetbrains.kotlin.plugin.serialization") version "1.4.32"
+}
+
+java {
+  sourceCompatibility = JavaVersion.VERSION_17
+  targetCompatibility = JavaVersion.VERSION_11
+}
+
+// The latest supported versions. Note, these are updated automatically from update-major-versions.sh
+val IIC_RELEASE_VERSION = "231.9011.34"
+val IIC_EAP_VERSION = "232.6095.10"
+val GO_RELEASE_VERSION = "231.9011.34"
+val GO_EAP_VERSION = "232.6095.10"
+// The oldest supported versions.
+val IIC_FROM_VERSION = "222.4554.10"
+val GO_FROM_VERSION = "222.4554.12"
+
+data class Product(
+  val name: String, // identifier for this product
+  val sdkVersion: String, // the version string passed to the intellij sdk gradle plugin
+  val goPluginVersion: String, // a specific version for the go plugin
+)
+
+val products = listOf(
+  Product(name = "release", sdkVersion = IIC_RELEASE_VERSION, goPluginVersion = IIC_RELEASE_VERSION),
+  Product(name = "eap", sdkVersion = "LATEST-EAP-SNAPSHOT", goPluginVersion = IIC_EAP_VERSION),
+)
+val product = products.first { it.name == (System.getenv("PRODUCT_NAME") ?: "eap") }
+
+val kotlin_version = "1.8.10"
+
+group = "com.squareup.cash.hermit"
+version = project.properties["version"] ?: "1.0-SNAPSHOT"
+
+tasks.withType<KotlinCompile> {
+  kotlinOptions.jvmTarget = "11"
+  kotlinOptions.freeCompilerArgs = listOf("-Xjvm-default=enable")
+}
+
+repositories {
+  mavenCentral()
+}
+
+dependencies {
+  implementation("org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version")
+  implementation("org.jetbrains.kotlin:kotlin-reflect:$kotlin_version")
+  implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.1.0")
+}
+
+intellij {
+  // Note: The IntelliJ version below needs to match the go plugin version as defined here:
+  // https://plugins.jetbrains.com/plugin/9568-go/versions
+  version.set(product.sdkVersion)
+  type.set("IU")
+  plugins.set(
+    listOf(
+      "gradle",
+      "java",
+      "terminal",
+      // This version is in sync with the IJ version, but not in sync with the GoLand version.
+      "org.jetbrains.plugins.go:${product.goPluginVersion}",
+      // Needed by Go plugin. See https://github.com/JetBrains/gradle-intellij-plugin/issues/1056
+      "org.intellij.intelliLang"
+    )
+  )
+}
+
+tasks.withType<RunIdeTask> {
+  // Uncomment this, and set your path accordingly, if you want to debug on GoLand
+  // ideDirectory "/Users/juho/Library/Application Support/JetBrains/Toolbox/apps/Goland/ch-0/203.6682.164/GoLand.app/Contents"
+}
+
+tasks.withType<PatchPluginXmlTask> {
+  sinceBuild.set(IIC_FROM_VERSION)
+  version.set(System.getenv("IJ_PLUGIN_VERSION"))
```

Comment:
what happens if this is null?

Reply:
It crashes. Both the patch and publish tasks are only expected to run in CI so that's not a big deal, but I should note that somewhere.